app/kubemci/cmd/version.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	versionDescription = "Prints the version of this tool"
+)
+
+const (
+	// String indicating the client version.
+	// x.y.z+ indicates that the client has some changes over the x.y.z release.
+	// Release builds should have versions of the form x.y.z.
+	// TODO(nikhiljindal): Add more useful information such as build date, git commit, etc similar to kubectl.
+	clientVersion = "0.0.0+"
+)
+
+func NewCmdGetVersion(out, err io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: versionDescription,
+		Long:  versionDescription,
+		// TODO(nikhiljindal): Add an example.
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := ValidateVersionArgs(args); err != nil {
+				fmt.Println(err)
+				return
+			}
+			if err := RunVersion(args); err != nil {
+				fmt.Println("Error in printing version:", err)
+			}
+		},
+	}
+	return cmd
+}
+
+func ValidateVersionArgs(args []string) error {
+	if len(args) != 0 {
```

Comment:
Why are ValidateVersionArgs and RunVersion functions and why are they exported?

Reply:
They are functions so that they can be tested individually.
Changed them to be private.

While at it, have also update similar methods at other places to be private.